src/main/java/seedu/address/model/issue/Issue.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.model.issue;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+/**
+ * Represents an Issue.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Issue {
+
+    // Components of an issue
+    private Id id;
+    private Description description;
+    private Deadline deadline;
+    private Priority priority;
+    private Status status;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Issue(Id id, Description description, Deadline deadline, Priority priority, Status status) {
+        requireAllNonNull(id, description, deadline, priority, status);
```

Comment:
Need to verify the fields that are compulsory and optional

Reply:
null checks for all parameters should be fine as discussed, but singleton classes should be present